### PR TITLE
Feature/backend/settings part2

### PR DESCRIPTION
**Pull request by gdchamal**
Adapt settings to new version as defined in US 363.

### CODE REVIEW

File: src/backend/main/py.main/caliopen_main/user/parameters/settings.py [hunk 1]
```diff
@@ -3,19 +3,29 @@
...
 from schematics.models import Model
-from schematics.types import StringType, IntType
+from schematics.types import StringType, IntType, BooleanType
+
+MESSAGE_FORMAT_CHOICES = ['rich_text', 'plain_text']
+CONTACT_FORMAT_CHOICES = ['given_name', 'family_name',
+                          'family_name, given_name']
+CONTACT_ORDER_CHOICES = ['family_name', 'given_name']
+PREVIEW_CHOICES = ['off', 'always']
+DELAY_CHOICES = [5, 10, 10, 30]
```
Comment (sapiens-sapide) [inline src/backend/main/py.main/caliopen_main/user/parameters/settings.py]: 5, 10, **20**, 30 may be ?

File: src/backend/main/py.main/caliopen_main/user/parameters/user.py [hunk 2]
```diff
@@ -27,6 +28,7 @@ class NewUser(Model):
     params = DictType(StringType())
     password = StringType(required=True)
     recovery_email = EmailType(required=True)
+    settings = ModelType(Settings, default=lambda: {})
```
Comment (sapiens-sapide) [inline src/backend/main/py.main/caliopen_main/user/parameters/user.py]: `default` could be a func that returns default settings from config file ?

Comment (gdchamal) [reply src/backend/main/py.main/caliopen_main/user/parameters/user.py]: default value are a mess and not handled correctly for the moment. default values are defined both in schematics Settings and in swagger object.

I think that I'll remove default value from schematics later and try to see how to handle only API side, ideally using a configuration file.

File: src/backend/main/py.main/caliopen_main/user/parameters/settings.py (hunk 1 above)
Comment (gdchamal) [reply src/backend/main/py.main/caliopen_main/user/parameters/settings.py]: I followed the spec ;)

File: src/backend/defs/rest-api/objects/Settings.yaml [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
 type: object
 properties:
-  default_language:
-    type: string
-  default_timezone:
-    type: string
-  date_format:
+  default_locale:
     type: string
+    default: fr_FR
```
Comment (iamdey) [inline src/backend/defs/rest-api/objects/Settings.yaml]: I was wrong, it should `fr-FR`, (this what I get from browser)

Comment (gdchamal) [reply src/backend/defs/rest-api/objects/Settings.yaml]: fixed in 8ab87b3879ea6a48d62968500bafd81f50d77915

**Pull request merged**